File: languages/csharp_unity_json_properties/Construct.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Construct : Element
{
    [JsonProperty("rationale"), TextAttribute("")]
    public string rationale;
    [JsonProperty("history"), TextAttribute("")]
    public string history;
    [JsonProperty("status"), TextAttribute("")]
    public string status;
    [JsonProperty("reach"), TextAttribute("")]
    public string reach;
    [JsonProperty("start_date"), Integer(0)]
    public int startDate;
    [JsonProperty("end_date"), Integer(0)]
    public int endDate;
    [JsonProperty("founder"), ReferenceAttribute(typeof(Character))]
    public string founder;
    [JsonProperty("custodian"), ReferenceAttribute(typeof(Institution))]
    public string custodian;
    [JsonProperty("characters"), ReferenceAttribute(typeof(Character), true)]
    public string characters;
    [JsonProperty("objects"), ReferenceAttribute(typeof(Object), true)]
    public string objects;
    [JsonProperty("locations"), ReferenceAttribute(typeof(Location), true)]
    public string locations;
    [JsonProperty("species"), ReferenceAttribute(typeof(Species), true)]
    public string species;
    [JsonProperty("creatures"), ReferenceAttribute(typeof(Creature), true)]
    public string creatures;
    [JsonProperty("institutions"), ReferenceAttribute(typeof(Institution), true)]
    public string institutions;
    [JsonProperty("traits"), ReferenceAttribute(typeof(Trait), true)]
    public string traits;
    [JsonProperty("collectives"), ReferenceAttribute(typeof(Collective), true)]
    public string collectives;
    [JsonProperty("territories"), ReferenceAttribute(typeof(Territory), true)]
    public string territories;
    [JsonProperty("abilities"), ReferenceAttribute(typeof(Ability), true)]
    public string abilities;
    [JsonProperty("phenomena"), ReferenceAttribute(typeof(Phenomenon), true)]
    public string phenomena;
    [JsonProperty("languages"), ReferenceAttribute(typeof(Language), true)]
    public string languages;
    [JsonProperty("families"), ReferenceAttribute(typeof(Family), true)]
    public string families;
    [JsonProperty("relations"), ReferenceAttribute(typeof(Relation), true)]
    public string relations;
    [JsonProperty("titles"), ReferenceAttribute(typeof(Title), true)]
    public string titles;
    [JsonProperty("constructs"), ReferenceAttribute(typeof(Construct), true)]
    public string constructs;
    [JsonProperty("events"), ReferenceAttribute(typeof(Event), true)]
    public string events;
    [JsonProperty("narratives"), ReferenceAttribute(typeof(Narrative), true)]
    public string narratives;
}

File: languages/django_models/construct.py
from .abstract_element_model import AbstractElementModel
from django.db import models
from ow.elements.models.object import Object as ObjectModel

class Construct(AbstractElementModel):

    # Nature
    rationale = models.TextField(blank=True, null=True)
    history = models.TextField(blank=True, null=True)
    status = models.TextField(blank=True, null=True)
    reach = models.TextField(blank=True, null=True)
    start_date = models.PositiveIntegerField(blank=True, null=True)
    end_date = models.PositiveIntegerField(blank=True, null=True)
    founder = models.ForeignKey("Character", on_delete=models.SET_NULL, blank=True, null=True, related_name="construct_founder")
    custodian = models.ForeignKey("Institution", on_delete=models.SET_NULL, blank=True, null=True, related_name="construct_custodian")

    # Involves
    characters = models.ManyToManyField("Character", blank=True, related_name="construct_characters")
    objects = models.ManyToManyField(ObjectModel, blank=True, related_name="construct_objects")  # type: ignore
    locations = models.ManyToManyField("Location", blank=True, related_name="construct_locations")
    species = models.ManyToManyField("Species", blank=True, related_name="construct_species")
    creatures = models.ManyToManyField("Creature", blank=True, related_name="construct_creatures")
    institutions = models.ManyToManyField("Institution", blank=True, related_name="construct_institutions")
    traits = models.ManyToManyField("Trait", blank=True, related_name="construct_traits")
    collectives = models.ManyToManyField("Collective", blank=True, related_name="construct_collectives")
    zones = models.ManyToManyField("Zone", blank=True, related_name="construct_zones")
    abilities = models.ManyToManyField("Ability", blank=True, related_name="construct_abilities")
    phenomena = models.ManyToManyField("Phenomenon", blank=True, related_name="construct_phenomena")
    languages = models.ManyToManyField("Language", blank=True, related_name="construct_languages")
    families = models.ManyToManyField("Family", blank=True, related_name="construct_families")
    relations = models.ManyToManyField("Relation", blank=True, related_name="construct_relations")
    titles = models.ManyToManyField("Title", blank=True, related_name="construct_titles")
    constructs = models.ManyToManyField("Construct", blank=True, related_name="construct_constructs")
    events = models.ManyToManyField("Event", blank=True, related_name="construct_events")
    narratives = models.ManyToManyField("Narrative", blank=True, related_name="construct_narratives")

    def __str__(self):
        return self.name
